disnake/partial_emoji.py

```diff
@@ -271,3 +272,40 @@ async def read(self) -> bytes:
             raise TypeError("PartialEmoji is not a custom emoji")
 
         return await super().read()
+
+    @staticmethod
+    def _emoji_to_name_id(
+        emoji: Optional[Union[str, Emoji, PartialEmoji]]
+    ) -> Tuple[Optional[str], Optional[int]]:
+        if emoji is None:
+            return None, None
+
+        if isinstance(emoji, str):
+            emoji = PartialEmoji.from_str(emoji)
+
+        # note: API only supports exactly one of `name` and `id` being set
+        if emoji.id:
+            return None, emoji.id
+        else:
+            return emoji.name, None
+
+    @staticmethod
+    def _emoji_from_name_id(
+        name: Optional[str], id: Optional[int], *, state: ConnectionState
+    ) -> Optional[Union[Emoji, PartialEmoji]]:
+        if not (name or id):
+            return None
+
+        emoji: Optional[Union[Emoji, PartialEmoji]] = None
+        if id:
+            emoji = state.get_emoji(id)
+        if not emoji:
+            emoji = PartialEmoji.with_state(
+                state=state,
+                # Note: this does not render correctly if it's a custom emoji, there's just no name information for those here.
+                # This may change in a future API version, but for now we'll just have to accept it.
```

Comment:
Decided to note this in the documentation instead of working around the name requirement: https://github.com/DisnakeDev/disnake/pull/726/commits/3248f5db2e318bdf0235ce867378bd74b3919a18
Generally this should only happen with deleted emojis, as it would've otherwise gotten the emoji from the cache already and skipped this part.